src/main/java/randoop/generation/constanttfidf/ConstantMiningStatistics.java

```diff
@@ -0,0 +1,298 @@
+package randoop.generation.constanttfidf;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import randoop.main.GenInputsAbstract;
+import randoop.main.RandoopBug;
+import randoop.sequence.Sequence;
+import randoop.types.ClassOrInterfaceType;
+import randoop.util.Log;
+
+/**
+ * This class stores constant mining information by the given literals level. T is the scope of the
+ * constant mining, which can be ClassOrInterfaceType, Package, or Object, which corresponds to
+ * users' input about literal level as CLASS, PACKAGE, or ALL. The scope statistics stores the
+ * information about constants within the given scope.
+ */
+public class ConstantMiningStatistics {
+
+  /**
+   * A map from a specific scope to its constant statistics. It contains each constant's number of
+   * times it is used, the number of classes it is contained, and the number of classes within the
+   * given scope.
+   */
+  private Map<Object, ScopeStatistics> scopeStatisticsMap;
+
+  /** Creates a ConstantMiningStatistics. */
+  public ConstantMiningStatistics() {
+    scopeStatisticsMap = new HashMap<>();
+  }
+
+  /**
+   * Add and update the frequency of the sequence to the current scope.
+   *
+   * @param type the type of the class
+   * @param seq the sequence to be added
+   * @param frequency the frequency of the sequence to be added
+   */
+  public void addUses(Object type, Sequence seq, int frequency) {
+    Object scope;
+    switch (GenInputsAbstract.literals_level) {
+      case CLASS:
+        scope = (ClassOrInterfaceType) type;
+        break;
+      case PACKAGE:
+        scope = ((ClassOrInterfaceType) type).getPackage();
+        break;
+      case ALL:
+        scope = null;
+        break;
+      default:
+        throw new RuntimeException("Unknown literals level");
```

Comment:
Abstract this out into a method, and use it in many places to make the code shorter.
Also, I suggest using a value other than null for the "ALL" scope, just to make nullness checking a bit easier.  You could even use the value `ALL`.